### ISSUE TITLE
forgeBSgenomeDataPkgFromNCBI() should raise an error if 'circ_seqs' is a mix of assembled molecules with non-assembled molecules

### ISSUE THREAD

**Issue by kakopo** [Bioconductor/BSgenomeForge#19]
In the event that a user specifies circular sequences that are a mix of assembled molecules with non-assembled molecules, the current `.get_circseqs()` helper function does not recognize that non-assembled molecules cannot be circular sequences. This can be seen with the example situation using assembly GCA_025782755.1:
```
library(BSgenomeForge)

seq_info <- getChromInfoFromNCBI("GCA_025782755.1")

seq_info[1:4 , c("SequenceName", "SequenceRole")]
#   SequenceName         SequenceRole
# 1          III   assembled-molecule
# 2           MT   assembled-molecule
# 3   CP088301.1 unlocalized-scaffold
# 4   CP088302.1 unlocalized-scaffold

BSgenomeForge:::.get_circseqs("GCA_025782755.1", seq_info, circ_seqs=c("MT", "CP088301.1"))
# [1] "MT"         "CP088301.1"
```
This issue must be rectified to raise an error message if non-assembled molecules are specified together with the assembled molecules.

**Comment by kakopo** [comment]
Hi @hpages. Please take a look at the most recent commit and let me know if it is okay.

**Comment by hpages** [comment]
Looks good but please double check the formatting of your `if` statement (positions of the curly brackets and indentation of the code within the statement)

        if (isEmpty(subset_seq_info)){
        return(circ_seqs)
        } else {
            if (! all(circ_seqs %in% assembled_molecules$SequenceName))
                stop(wmsg("the sequence names in 'circ_seqs' must be the names
                          of assembled molecules"))
        return(circ_seqs) }

BTW is the `isEmpty(subset_seq_info)` test necessary?

Thanks

**Comment by kakopo** [comment]
Alright. I have removed the `isEmpty(subset_seq_info)` check (did not affect the code in any way), and improved the editing. Please take a look at the latest commit. Thank you.

**Comment by hpages** [comment]
Thanks.

I just merged PR #21 so we can close this. Please remember to resync your fork. 

Next would be issue #16. Then we'll discuss unit tests for `BSgenomeForge:::.get_circseqs()` (issue #17).

H.
